oracle/tests/test_oracle.py

```diff
@@ -1,288 +1,310 @@
+
+        # No metric sent if errors
+        gauge.assert_not_called()
+
+    def test__get_custom_metrics(self, aggregator, check):
+        con = mock.MagicMock()
+        cursor = mock.MagicMock()
+        cursor.fetchone.side_effect = [
+            ["tag_value1", "1"],
+            [1, 2, "tag_value2"]
+        ]
+        con.cursor.return_value = cursor
+
+        custom_queries = [
+            {
+                "metric_prefix": "oracle.test1",
+                "query": "mocked",
+                "columns": [
+                    {
+                        "name": "tag_name",
+                        "type": "tag"
+                    },
+                    {
+                        "name": "metric",
+                        "type": "gauge"
+                    }
+                ],
+                "tags": ["query_tags1"]
+            },
+            {
+                "metric_prefix": "oracle.test2",
+                "query": "mocked",
+                "columns": [
+                    {
+                        "name": "rate",
+                        "type": "rate"
+                    },
+                    {
+                        "name": "gauge",
+                        "type": "gauge"
+                    },
+                    {
+                        "name": "tag_name",
+                        "type": "tag"
+                    }
+                ],
+                "tags": ["query_tags2"]
+            }
+        ]
+
+        check._get_custom_metrics(con, custom_queries, ["custom_tag"])
+        aggregator.assert_metric("oracle.test1.metric", value=1, count=1,
+                                 tags=["tag_name:tag_value1", "query_tags1", "custom_tag"])
+        aggregator.assert_metric("oracle.test2.gauge", value=2, count=1, metric_type=aggregator.GAUGE,
+                                 tags=["tag_name:tag_value2", "query_tags2", "custom_tag"])
+        aggregator.assert_metric("oracle.test2.rate", value=1, count=1, metric_type=aggregator.RATE,
+                                 tags=["tag_name:tag_value2", "query_tags2", "custom_tag"])
+
+
+class TestMetrics:
+    def test__get_sys_metrics(self, aggregator, check):
+        query = "SELECT METRIC_NAME, VALUE, BEGIN_TIME FROM GV$SYSMETRIC ORDER BY BEGIN_TIME"
+        con = mock.MagicMock()
+        cur = mock.MagicMock()
+        con.cursor.return_value = cur
+        cur.fetchall.return_value = zip(check.SYS_METRICS.keys(), [0] * len(check.SYS_METRICS.keys()))
+
+        check._get_sys_metrics(con, ["custom_tag"])
+
+        cur.execute.assert_called_with(query)
+        for _, metric in check.SYS_METRICS.items():
+            aggregator.assert_metric(metric, count=1, value=0, tags=["custom_tag"])
+
+    def test__get_process_metrics(self, aggregator, check):
+        query = "SELECT PROGRAM, {} FROM GV$PROCESS".format(','.join(check.PROCESS_METRICS.keys()))
```

Comment:
This seems kinda sketchy, I feel like I should be hardcoding this string.

Reply:
you can if you want!